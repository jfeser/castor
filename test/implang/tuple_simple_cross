ATuple([AScalar(1), AScalar(2)], cross)