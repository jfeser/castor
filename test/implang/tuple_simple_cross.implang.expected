// Locals:
// cstart3 : int (persists=true)
// cstart2 : int (persists=true)
fun printer () : void {
    cstart2 = 0;
    cstart3 = cstart2 + 1;
    print(tuple[int, int], (buf[cstart2 : 1], buf[cstart3 : 1]));
}
// Locals:
// cstart1 : int (persists=true)
// cstart0 : int (persists=true)
fun consumer () : void {
    cstart0 = 0;
    cstart1 = cstart0 + 1;
    consume(tuple[int, int], (buf[cstart0 : 1], buf[cstart1 : 1]));
}
